src/util/errors/HttpError.ts

```diff
@@ -1,23 +1,32 @@
 import { isError } from './ErrorUtil';
 
+export interface HttpErrorOptions {
+  cause?: unknown;
+  errorCode?: string;
+  params?: NodeJS.Dict<unknown>;
+}
+
 /**
  * A class for all errors that could be thrown by Solid.
  * All errors inheriting from this should fix the status code thereby hiding the HTTP internals from other components.
  */
 export class HttpError extends Error {
   protected static readonly statusCode: number;
   public readonly statusCode: number;
+  public readonly options: HttpErrorOptions;
 
   /**
    * Creates a new HTTP error. Subclasses should call this with their fixed status code.
    * @param statusCode - HTTP status code needed for the HTTP response.
    * @param name - Error name. Useful for logging and stack tracing.
-   * @param message - Message to be thrown.
+   * @param message - Error message.
+   * @param options - Optional options.
    */
-  public constructor(statusCode: number, name: string, message?: string) {
+  public constructor(statusCode: number, name: string, message?: string, options: HttpErrorOptions = {}) {
```

Comment:
Shall we get `statusCode` and `name` also into the options hash?

Reply:
Some issues with that:
 * statusCode and name are not optional (name could be made optional though), so the options object would no longer be optional.
 * Can't re-use the interface since the specific error classes already have a set name and status code.

There are some ways around this with extra interfaces but I would just leave it as is since HttpError is mostly used as an abstract class. The constructor only gets used directly when we're aggregating errors in the waterfall handler.

Reply:
It seems the first issue is mitigated by it being an abstract class; I don't understand the second issue.

But not a big thing; just the only place where message and options are not the first two arguments.

Reply:
What I meant was that the constructor of all instanced errors is `constructor(message?: string, options?: HttpErrorOptions)`. If status code also ends up in the options interface, we would no longer be able to do the following: `new BadRequestHttpError('msg', { cause: error })` since we are not providing a value for the required value `statusCode` in the options object. We could have a separate interface that does not contain statusCode and name, and then extend that for the input options required in the base class.

That and the linter doesn't want us to put optional arguments before required ones so we would have to switch the order or tell the linter to shut up there. ;)

Reply:
> Yeah, I meant that the (only) the base class has `HttpErrorOptions | { statusCode: number, name: string }`

Ah like that, yea that would work (after adding a lint ignore option).